client/src/components/FormField/FileUpload.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import styles from './fileUpload.module.scss';
+
+interface OwnProps {
+  title?: string;
+  note?: string;
+  containerClass?: string;
+  onDelete: () => void;
+}
+
+type FileUploadProps = OwnProps & React.InputHTMLAttributes<HTMLInputElement>;
+
+export const FileUpload: React.FC<FileUploadProps> = ({
+  containerClass = '',
+  title,
+  note,
+  onChange,
+  id,
+  name,
+  onDelete,
+  ...inputProps
+}) => {
+  const [fileName, setFileName] = React.useState<string>('');
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  if (!id || !name) {
+    throw new Error('Id and name for file input has to be passed!');
+  }
+
+  const Title = title && <span className={styles.title}>{title}</span>;
+  const Note = note && <span className={styles.note}>{note}</span>;
+
+  const classNameWrapper = classNames(styles.customInput, containerClass, {
+    [styles.customInputFileSelected]: fileName,
+  });
+  const classNameButton = classNames(styles.button, {
+    [styles.buttonFileSelected]: fileName,
+  });
+
+  const handleUploadButtonClick = (): void => {
+    if (fileName) onDelete();
+    if (!fileName) inputRef.current?.click();
+  };
+
+  React.useEffect(() => {
+    if (inputRef.current) {
+      const { files } = inputRef.current;
+      const name = files ? (files[0] ? files[0].name : '') : '';
+      setFileName(name);
+    }
```

Comment:
```suggestion
    if (inputRef.current) {
      const { files = [] } = inputRef.current;
      const name = files.length ? files[0].name : '';
      setFileName(name);
    }
```

Reply:
still ts type error so
```suggestion
    if (inputRef.current) {
      const { files = [] } = inputRef.current;
      const name = files && files.length ? files[0].name : '';
      setFileName(name);
    }
```